kernel/src/main/java/org/kframework/compile/checks/CheckFunctions.java

```diff
@@ -83,4 +85,26 @@ public void apply(KApply k) {
             }
         }.apply(body);
     }
+
+    public void checkFuncAtt(RuleOrClaim r) {
+        new RewriteAwareVisitor(true, errors) {
+            @Override
+            public void apply(KApply k) {
+                if (isRHS() || k.klabel() instanceof KVariable || !m.attributesFor().contains(k.klabel())) {
```

Comment:
Instead of checking isRHS, which will return true outside of any rewrite, you probably want to check `isRHS() && !isLHS()`